node/pkg/algorand/watcher.go

```diff
@@ -2,29 +2,250 @@ package algorand
+			if e.debug {
+				logger.Info("emitter: " + hex.EncodeToString(emitter[:]))
+			}
+
+			id, err := base32.StdEncoding.WithPadding(base32.NoPadding).DecodeString(t.Id)
+			if nil != err {
+				logger.Info(err.Error())
+				continue;
+			}
+
+			if e.debug {
+				logger.Info("id: " + hex.EncodeToString(id) + " " + t.Id)
+			}
+
+			var txHash = eth_common.BytesToHash(id)   // 32 bytes = d3b136a6a182a40554b2fafbc8d12a7a22737c10c81e33b33d1dcb74c532708b
+
+			observation := &common.MessagePublication{
+				TxHash:           txHash,
+				Timestamp:        time.Unix(int64(it.RoundTime), 0),
+				Nonce:            uint32(binary.BigEndian.Uint64(at.ApplicationArgs[2])),
+				Sequence:         binary.BigEndian.Uint64(it.Logs[0]),
+				EmitterChain:     vaa.ChainIDAlgorand,
+				EmitterAddress:   a,
+				Payload:          at.ApplicationArgs[1],
+				ConsistencyLevel: 32,   // What SHOULD this be?
+			}
+			
+			algorandMessagesConfirmed.Inc()
+
+			logger.Info("message observed",
+				zap.Time("timestamp", observation.Timestamp),
+				zap.Uint32("nonce", observation.Nonce),
+				zap.Uint64("sequence", observation.Sequence),
+				zap.Stringer("emitter_chain", observation.EmitterChain),
+				zap.Stringer("emitter_address", observation.EmitterAddress),
+				zap.Binary("payload", observation.Payload),
+				zap.Uint8("consistency_level", observation.ConsistencyLevel),
+			)
+
+			e.msgChan <- observation
+		}
+	}
 }
 
 func (e *Watcher) Run(ctx context.Context) error {
-	readiness.SetReady(common.ReadinessAlgorandSyncing)
+	// an odd thing to broadcast... 
+	p2p.DefaultRegistry.SetNetworkStats(vaa.ChainIDAlgorand, &gossipv1.Heartbeat_Network{
+		ContractAddress: fmt.Sprintf("%d", e.appid),
+	})
+
+	logger := supervisor.Logger(ctx)
+	errC := make(chan error)
+
+	logger.Info("Algorand watcher connecting", zap.String("url", e.indexerRPC))
+
+	go func() {
+		timer := time.NewTicker(time.Second * 1)
+		defer timer.Stop()
+
+		indexerClient, err := indexer.MakeClient(e.indexerRPC, e.indexerToken)
+		if err != nil {
+			logger.Info(err.Error())
+			p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDAlgorand, 1)
+			errC <- err
+			return
+		}
+
+		// Parameters
+		var notePrefix = "publishMessage"
 
-	select {}
+		if e.next_round == 0 {
+			// What is the latest round...
+			result, err := indexerClient.SearchForTransactions().Limit(0).Do(context.Background())
+			if err != nil {
+				logger.Info(err.Error())
+				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDAlgorand, 1)
+				errC <- err
+				return
+			}
+			e.next_round = result.CurrentRound + 1
+			logger.Info("Algorand next_round set to " + fmt.Sprintf("%d", e.next_round))
+		}
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case r := <-e.obsvReqC:
+				if vaa.ChainID(r.ChainId) != vaa.ChainIDAlgorand {
+					panic("invalid chain ID")
+				}
+
+				logger.Info("Received obsv request: " + hex.EncodeToString(r.TxHash) + " -> " + base32.StdEncoding.WithPadding(base32.NoPadding).EncodeToString(r.TxHash))
```

Comment:
Can you make this look more consistent with the other ones like:

https://github.com/certusone/wormhole/blob/5a987a8d2bb90d6fb294455780e641bf1d784b9d/node/pkg/ethereum/watcher.go#L222-L224

Grepping through logs where some chains shorten it and some don't is super annoying for operators. Please, think of the poor folks running these guardians. If you don't, they'll look like this guy:

![wat](https://media.giphy.com/media/kc0kqKNFu7v35gPkwB/giphy.gif)

Reply:
The information I need is different and that does not contain the information I wanted.